source/php/Component/MegaMenu/megaMenu.blade.php

```diff
@@ -50,28 +50,46 @@ class="{{$baseClass}}__item {{$baseClass}}__item--parent o-grid-12 o-grid-6@md o
-                                @link([
-                                    'href' => $child['href'],
-                                    'classList' => [
-                                        $baseClass . '__link', 
-                                        $baseClass . '__link--child'
-                                    ]
-                                ])
-                                @if(isset($child['icon']) && !array_diff(['icon', 'size', 'classList'], array_keys($child['icon'])))
-                                    @icon([
-                                        'icon' => $child['icon']['icon'],
-                                        'size' => $child['icon']['size'],
-                                        'classList' => $child['icon']['classList'],
+                                @if(!empty($childStyle))
+                                    @button([
+                                        'text' => $child['label'],
+                                        'style' => $childStyle,
+                                        'color' => $childStyleColor ?? 'primary',
+                                        'shape' => $childStyleShape ?? 'pill',
+                                        'icon' => $child['icon']['icon'] !== "" ? $child['icon']['icon'] : 'chevron_right',
+                                        'href' => $child['href'],
+                                        'size' => 'sm',
+                                        'classList' => [
+                                            $baseClass . '__link',
+                                            $baseClass . '__link--button',
+                                        ]
+                                    ])
+                                    @endbutton
+                                @else
+                                    @link([
+                                        'href' => $child['href'],
+                                        'classList' => [
+                                            $baseClass . '__link', 
+                                            $baseClass . '__link--child'
+                                        ]
                                     ])
-                                    @endicon
+                                    @if(isset($child['icon']) && !array_diff(['icon', 'size', 'classList'], array_keys($child['icon'])))
+                                        @icon([
+                                            'icon' => $child['icon']['icon'],
+                                            'size' => $child['icon']['size'],
+                                            'classList' => $child['icon']['classList'],
+                                        ])
+                                        @endicon
+                                    @endif
+                                        {{ $child['label'] }}
+                                    @endlink
```

Comment:
The different child styles can probably be put in partials to make reading and adding more styles easier in the future